pandas/io/parquet.py

```diff
@@ -115,17 +118,21 @@ def write(
                 **kwargs,
             )
         else:
-            self.api.parquet.write_table(
-                table, file_obj_or_path, compression=compression, **kwargs
-            )
-        if should_close:
-            file_obj_or_path.close()
+            # write to single output file
+            self.api.parquet.write_table(table, path, compression=compression, **kwargs)
 
     def read(self, path, columns=None, **kwargs):
-        fs = get_fs_for_path(path)
-        should_close = None
-        # Avoid calling get_filepath_or_buffer for s3/gcs URLs since
-        # since it returns an S3File which doesn't support dir reads in arrow
+        if is_fsspec_url(path) and "filesystem" not in kwargs:
+            import_optional_dependency("fsspec")
+            import fsspec.core
+
+            fs, path = fsspec.core.url_to_fs(path)
+            should_close = False
+        else:
+            fs = kwargs.pop("filesystem", None)
+            should_close = False
+            path = _expand_user(path)
```

Comment:
No blocking comment (could also be as a follow-up), but for my understanding (and could maybe use a code comment): what was the reason again that this whole `is_fsspec_url(path): .. else: ..` block cannot be handled by `get_filepath_or_buffer` ? 
(which would eg handle the `_expand_user` as well)

Reply:
That we are not getting an open file, we are passing a path and filesystem to arrow for it to handle. This is important for the partitioning case, but works for the simple case too. Wasn't it your suggestion?

Reply:
> Wasn't it your suggestion?

I don't think so, since this was already there from the beginning of the PR before my first review ;) Not that it matters much.

Anyway, to answer my own question (I think): on a second look, this is simply because `get_filepath_or_buffer` doesn't return a filesystem (and which is indeed needed here, as you indicated). I was just confused about that for a moment.